File: app/src/main/java/com/snaptiongame/app/data/providers/SessionProvider.java
package com.snaptiongame.app.data.providers;

import com.snaptiongame.app.data.api.SnaptionApi;
import com.snaptiongame.app.data.models.OAuthRequest;
import com.snaptiongame.app.data.models.Session;
import com.snaptiongame.app.data.providers.api.ApiProvider;

import io.reactivex.Completable;
import io.reactivex.Single;

/**
 * @author Tyler Wong
 */

public class SessionProvider {
    private static SnaptionApi apiService = ApiProvider.getApiService();

    public static Single<Session> userOAuthFacebook(OAuthRequest request) {
        return apiService.userOAuthFacebook(request);
    }

    public static Single<Session> userOAuthGoogle(OAuthRequest request) {
        return apiService.userOAuthGoogle(request);
    }

    public static Completable logout() {
        return apiService.logout();
    }
}

File: app/src/main/java/com/snaptiongame/app/data/providers/SessionProvider.kt
@file:JvmName("SessionProvider")

package com.snaptiongame.app.data.providers

import com.snaptiongame.app.data.models.OAuthRequest
import com.snaptiongame.app.data.models.Session
import com.snaptiongame.app.data.providers.api.ApiProvider

import io.reactivex.Completable
import io.reactivex.Single

/**
 * @author Tyler Wong
 */

private val apiService = ApiProvider.getApiService()

fun userOAuthFacebook(request: OAuthRequest): Single<Session> = apiService.userOAuthFacebook(request)

fun userOAuthGoogle(request: OAuthRequest): Single<Session> = apiService.userOAuthGoogle(request)

fun logout(): Completable = apiService.logout()

fun isSessionValid(): Single<Boolean> = apiService.isSessionValid()
